app/validators/claim/advocate_claim_validator.rb

```diff
@@ -72,11 +75,21 @@ def validate_defendant_uplifts
 
   # we add one because uplift quantities reflect the number of "additional" defendants
   def defendant_uplifts_greater_than?(no_of_defendants)
-    @record
-      .fees
-      .where.not(id: @record.misc_fees.select(&:marked_for_destruction?).map(&:id))
-      .defendant_uplift_sums
-      .values
-      .map(&:to_i).any? { |sum| sum + 1 > no_of_defendants }
+    defendant_uplifts_counts.map(&:to_i).any? { |sum| sum + 1 > no_of_defendants }
+  end
+
+  def defendant_uplifts
```

Comment:
📝 This piece of code was refactored as well, mainly because the way this validation was being performed was to retrieve all the _database_ **fees** associated with the claim.

What I found out once I've separated the fees pages was that when I was submitting the **fixed fees** they were assigned (in memory) to the `fixed_fees` association on the claim, so calling `fees` on the claim was returning only the pre-setup basic fees and not the fixed fees.

This new approach retrieves all the types of fees from memory, while maintaining the same check to verify the number of uplifts.

Reply:
does it handle fixed fees that are marked for destruction?

Reply:
it does, just saw that below!